core/pkg/sync/http/http_sync.go

```diff
@@ -9,25 +9,79 @@ import (
 	"io"
 	"net/http"
 	parseUrl "net/url"
+	"os"
 	"path/filepath"
+	"time"
 
 	"github.com/open-feature/flagd/core/pkg/logger"
 	"github.com/open-feature/flagd/core/pkg/sync"
 	"github.com/open-feature/flagd/core/pkg/utils"
+	"github.com/robfig/cron"
+	"go.uber.org/zap"
 	"golang.org/x/crypto/sha3" //nolint:gosec
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
 )
 
 type Sync struct {
-	URI         string
-	Client      Client
-	Cron        Cron
-	LastBodySHA string
-	Logger      *logger.Logger
-	BearerToken string
-	AuthHeader  string
-	Interval    uint32
+	uri         string
+	client      Client
+	cron        Cron
+	lastBodySHA string
+	logger      *logger.Logger
+	bearerToken string
+	authHeader  string
+	interval    uint32
 	ready       bool
 	eTag        string
+
+	oauthCredential *oauthCredentialHandler
+}
+
+type oauthCredentialHandler struct {
+	clientId           string
+	clientSecret       string
+	tokenUrl           string
+	folderSource       string
+	reloadDelaySeconds int
+
+	lastUpdate time.Time
+}
+
+func (och *oauthCredentialHandler) loadOAuthConfiguration() (*clientcredentials.Config, error) {
+	if och.folderSource == "" {
```

Comment:
we should properly document, that if a foldersource is specified, we prioritize this, and ignore the definied clientId and ClientSecret - based on https://github.com/open-feature/flagd/pull/1791/files#r2332321164 - anyways we should properly document it, based on the outcome of the other comment

Reply:
yes - doc will be done in a follow-up PR. I have currently issues with the build of the docs locally :/ something is not supported by ARM64